explorer-backend/transactions/evm.go

```diff
@@ -151,47 +151,6 @@ func (w *EVMWatcher) fetchEventsFromBlockRange(ctx context.Context, fromHeight u
 
 func (w *EVMWatcher) fetchEventsByBlockHash(ctx context.Context, blockHash ethCommon.Hash) error {
 	client := w.connector.Client()
-	header, err := client.HeaderByHash(ctx, blockHash)
-	if err != nil {
-		w.logger.Error("failed to get header by block hash", zap.String("blockHash", blockHash.Hex()), zap.Error(err))
-		return err
-	}
-	return w.fetchEventsByHeader(ctx, header)
-}
-
-func (w *EVMWatcher) bridgeTxFromLog(ctx context.Context, client *ethclient.Client, log *types.Log) (*BridgeTransaction, error) {
-	message, err := w.connector.ParseLogMessagePublished(*log)
-	if err != nil {
-		w.logger.Error("failed to parse message", zap.String("txId", log.TxHash.Hex()), zap.Error(err))
-		return nil, err
-	}
-	if !isTransferTokenPayload(message.Payload) {
-		w.logger.Debug("ignore non-transfer token message", zap.String("txId", log.TxHash.Hex()))
-		return nil, nil
-	}
-	vaaId := &vaa.VAAID{
-		EmitterChain:   w.chainId,
-		EmitterAddress: ethereum.PadAddress(message.Sender),
-		TargetChain:    vaa.ChainID(message.TargetChainId),
-		Sequence:       message.Sequence,
-	}
-	sender, err := getEVMTxSender(ctx, client, log.TxHash)
-	if err != nil {
-		w.logger.Error("failed to get tx sender", zap.String("txId", log.TxHash.Hex()), zap.Error(err))
-		return nil, err
-	}
-	w.logger.Info("new bridge transaction", zap.String("sender", *sender), zap.String("txId", log.TxHash.Hex()))
-	return &BridgeTransaction{
-		vaaId:      vaaId,
-		txId:       log.TxHash.Hex(),
-		address:    *sender,
-		eventIndex: uint32(log.BlockNumber),
-	}, nil
-}
-
-func (w *EVMWatcher) fetchEventsByHeader(ctx context.Context, header *types.Header) error {
```

Comment:
We could pass the client as function argument to make the diff smaller

Reply:
I have already merged these two functions, so we don't need to pass the client as a parameter. Actually I made very few code changes, I simply moved a few lines of code. The extensive diff here is mainly because the `bridgeTxFromLog` function is located between these two functions, but the `bridgeTxFromLog` function remains unchanged.

Reply:
Okay